jpo-ode-svcs/src/main/java/us/dot/its/jpo/ode/services/asn1/Asn1EncodedDataRouter.java

```diff
@@ -404,4 +419,32 @@ private String stripHeader(String encodedUnsignedTim) {
       toReturn = encodedUnsignedTim.substring(index);
       return toReturn;
    }
+
+   private Void depositToFilteredTopic(JSONObject metadataObj, String hexEncodedTim, ServiceRequest request) {
+      try {
+         String generatedBy = metadataObj.getString("recordGeneratedBy");
+         String streamId = metadataObj.getJSONObject("serialId").getString("streamId");
+         if (generatedBy.equalsIgnoreCase("TMC")) {
+            try {
+               String timString = odeTimJsonTopology.query(streamId);
+
+               if (timString != null) {
+                  // Set ASN1 data in TIM metadata
+                  JSONObject timJSON = new JSONObject(timString);
+                  JSONObject metadataJSON = timJSON.getJSONObject("metadata");
+                  metadataJSON.put("asn1", hexEncodedTim);
+                  timJSON.put("metadata", metadataJSON);
+
+                  // Send the message w/ asn1 data to the TMC-filtered topic
+                  stringMsgProducer.send(odeProperties.getKafkaTopicOdeTimJsonTMCFiltered(), null, timJSON.toString());
+               }
+            } catch (Exception e) {
+               logger.error("Error while updating TIM: {}", e.getMessage());
+            }
+         }
+      } catch (Exception e) {
+         logger.error("Error while fetching recordGeneratedBy field: {}", e.getMessage());
+      }
+      return null;
```

Comment:
Is returning 'null' here necessary since the method has a return type of void? Also, should the return type be capitalized like it is currently?

Reply:
Swapped from "Void" to "void" and removed return - good catch!